v-next/core/test/internal/user-interruptions/user-interruptions-manager.ts

```diff
@@ -0,0 +1,111 @@
+import assert from "node:assert/strict";
+import { describe, it } from "node:test";
+
+import { HookManagerImplementation } from "../../../src/internal/hook-manager.js";
+import { UserInterruptionManagerImplementation } from "../../../src/internal/user-interruptions.js";
+import { UserInterruptionHooks } from "../../../src/types/hooks.js";
+
+describe("UserInterruptionManager", () => {
+  describe("displayMessage", () => {
+    it("Should call a dynamic handler with a given message from an interruptor", async () => {
+      const hookManager = new HookManagerImplementation([]);
+      const userInterruptionManager = new UserInterruptionManagerImplementation(
+        hookManager,
+      );
+      hookManager.setContext({} as any);
+
+      let called = false;
+      let givenInterruptor: string = "";
+      let givenMessage: string = "";
+
+      const handlers: Partial<UserInterruptionHooks> = {
+        async displayMessage(_context, interruptor, message) {
+          called = true;
+          givenInterruptor = interruptor;
+          givenMessage = message;
+        },
+      };
+
+      hookManager.registerHandlers("userInterruptions", handlers);
+
+      await userInterruptionManager.displayMessage(
+        "test-interruptor",
+        "test-message",
+      );
+
+      assert(called);
```

Comment:
It's better if we add a `message` to `assert`, as it's an alias of `assert.ok` and there's a bug that makes `assert.ok` really slow unless you provide a `message`. See: https://github.com/nodejs/node/issues/52962